src/com/cdd/bao/axioms/CellLineFix.java

```diff
@@ -0,0 +1,582 @@
+	{
+		// term with a label containing the keyword "cell" by defualt is re-parented to the "BRENDA cell parent" in CLO
+		// otherwise, term is re-parented to the "BRENDA" tissue parent" in CLO 
+
+		Pattern pat = Pattern.compile("\\s+cell\\s+");
+		Matcher mat = pat.matcher(label);
+
+		// create artificial branches on-demand for BRENDA cell and tissue terms
+		if (mat.matches() && !cloMap.containsKey(URI_FOR_CELL_LINES))
+		{
+			String branchLabel = "BRENDA cell lines";
+
+			outWriter.println("# create BRENDA cell branch and make its parent the CLO root");
+			outWriter.println(URI_FOR_CELL_LINES + " rdfs:label \"" + branchLabel + "\" ;");
+			outWriter.println("\trdfs:subClassOf " + URI_CLO_ROOT + " .");
+
+			cloMap.put(URI_FOR_CELL_LINES, branchLabel);
+		}
+		else if (!mat.matches() && !cloMap.containsKey(URI_FOR_TISSUES))
+		{
+			String branchLabel = "BRENDA tissue lines";
+
+			outWriter.println("# create BRENDA tissue branch and make its parent the CLO root");
+			outWriter.println(URI_FOR_TISSUES + " rdfs:label \"" + branchLabel + "\" ;");
+			outWriter.println("\trdfs:subClassOf " + URI_CLO_ROOT + " .");
+
+			cloMap.put(URI_FOR_TISSUES, branchLabel);
+		}
+
+		String uriPfx = ModelSchema.collapsePrefix(uriExp);
+		String brendaDesc = vocab.getDescr(uriExp);
+		String end = !StringUtils.isEmpty(brendaDesc) ? ";" : ".";
+
+		if (mat.matches())
+		{
+			outWriter.println("# re-parent BRENDA cell term to CLO branch");
+			outWriter.println(uriPfx + " bat:notSubClass <" + URI_BRENDA_ROOT + "> ;");
+			outWriter.println("\trdfs:subClassOf " + URI_FOR_CELL_LINES + " " + end);
+		}
+		else
+		{
+			outWriter.println("# re-parent BRENDA tissue term to CLO branch");
+			outWriter.println(uriPfx + " bat:notSubClass <" + URI_BRENDA_ROOT + "> ;");
+			outWriter.println("\trdfs:subClassOf " + URI_FOR_TISSUES + " " + end);
+		}
+
+		if (!StringUtils.isEmpty(brendaDesc))
+			outWriter.println("\tobo:IAO_0000115 \"" + brendaDesc + "\" .");
+
+		outWriter.println(""); // trailing blank line
+	}
+
+	// uri should be fully expanded
+	private void handleUnmatchedCLOTerms(NavigableMap<String, String> unmatchedCLOTerms)
+	{
+		outWriter.println("\n################################################################################");
+		outWriter.println("# following CLO terms are missing descriptions");
+		outWriter.println("# please correct and uncomment as needs be");
+		outWriter.println("################################################################################\n");
+
+		for (String uriExp : unmatchedCLOTerms.keySet())
+		{
+			String cloDesc = vocab.getDescr(uriExp);
+			if (StringUtils.isEmpty(cloDesc))
+			{
+				String uriPfx = ModelSchema.collapsePrefix(uriExp);
+				if (StringUtils.equals(uriPfx, uriExp))
+				{
+					// this obscure case happens when uri cannot be abbreviated
+					outWriter.println("# <" + uriPfx + "> obo:IAO_0000115 \"TBD: term description goes here\" .");
```

Comment:
Make the "TBD" thing blank, because some lucky person is going to have to type things into that space.

Reply:
Also: it would probably be helpful to add a preceding line that has a reminder of what the label is, and perhaps the hierarchy too, e.g.

```# cell line cell -> immortal cell line cell -> AVA5 cell
# obo:CLO_0037247 obo:IAO_0000115 "" .```